dags/naver_villa_real_estate.py

```diff
@@ -0,0 +1,415 @@
+from datetime import datetime, timedelta, date
+from io import StringIO
+
+import pandas as pd
+import requests
+import logging
+from airflow.decorators import task, dag, task_group
+from airflow.models import Variable
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+from airflow.providers.amazon.aws.hooks.redshift_sql import RedshiftSQLHook
+
+default_args = {
+    'owner': 'yong',
+    'depends_on_past': False,
+    'email_on_failure': False,
+    'email_on_retry': False,
+    'retries': 1,
+    'retry_delay': timedelta(seconds=30),
+}
+
+iam_role = Variable.get('aws_iam_role')
+
+
+def get_redshift_connection(autocommit=True):
+    hook = RedshiftSQLHook(redshift_conn_id='redshift_dev')
+    conn = hook.get_conn()
+    conn.autocommit = autocommit
+    return conn.cursor()
+
+
+# file_name 파일이 S3 bucket_name 에 있는가?
+def is_check_s3_file_exists(bucket_name, file_name):
+    s3_hook = S3Hook(aws_conn_id='aws_s3_connection')
+    key = 'data/' + file_name
+    return s3_hook.check_for_key(key=key, bucket_name=bucket_name)
+
+
+# s3의 data/ 폴더의 file_name CSV 파일 가져오기
+def get_csv_from_s3(bucket_name, file_name):
+    s3_hook = S3Hook(aws_conn_id='aws_s3_connection')
+    key = 'data/' + file_name
+    return pd.read_csv(s3_hook.get_key(key=key, bucket_name=bucket_name).get()['Body'])
+
+
+# 법정동코드(cortarNo) 지역의 동네 리스트를 데이터프레임으로 가져오기
+def get_region_info(url, headers, cortarNo):
+    url = f"{url}{cortarNo}"
+    res = requests.get(url, headers=headers)
+    _json = res.json()
+    return pd.DataFrame(_json["regionList"])
+
+
+# 데이터프레임을 S3에 CSV파일로 업로드
+def upload_to_s3(bucket_name, file_name, data_frame):
+    s3_hook = S3Hook(aws_conn_id='aws_s3_connection')
+    csv_buffer = StringIO()
+    key = 'data/' + file_name
+    data_frame.to_csv(csv_buffer, index=False)
+    s3_hook.load_string(string_data=csv_buffer.getvalue(), key=key, bucket_name=bucket_name, replace=True)
+
+
+# 법정동 코드로 해당 동네 빌라/주택 매물 리스트 검색
+def get_villa_real_estate(headers, cortarNo):
+    url = "https://new.land.naver.com/api/articles"
+    params = {
```

Comment:
상수값은 제일 상단에 대문자로 빼주세요.
```suggestion
    URL = "https://new.land.naver.com/api/articles"
```
상수값을 되도록 쓰지 않는 것이 제일 좋지만 써야한다면, 제일 상단으로 올려서 쉽게 인지/수정할 수 있도록 해야합니다.